packages/react/src/exampleTypes/ExpandableTagsQuery/index.js

```diff
@@ -7,36 +7,142 @@ import ExpandArrow from './ExpandArrow.js'
+    />
+  </svg>
+)
+
+let ExpandableTagsQuery = ({
+  measureRef,
+  contentRect,
+  collapse,
+  theme,
+  tree,
+  node,
+  Loader,
+  ...props
+}) => {
+  let generationsCollapse = React.useState(true)
 
-let ExpandableTagsQuery = ({ measureRef, contentRect, collapse, ...props }) => (
-  <>
+  let ref = React.useRef()
+  useOutsideClick({ ref, handler: F.on(generationsCollapse) })
+  return (
     <div
-      style={{
-        overflow: 'hidden',
-        maxHeight: F.view(collapse) ? innerHeightLimit : '',
+      ref={ref}
+      onMouseUp={(e) => {
+        e.stopPropagation()
       }}
     >
-      <div ref={measureRef}>
-        <TagsWrapper {..._.omit('measure', props)} />
+      <div>
+        <div
+          style={{
+            overflow: 'hidden',
+            maxHeight: F.view(collapse) ? innerHeightLimit : '',
+          }}
+        >
+          <div ref={measureRef}>
+            <TagsWrapper
+              {..._.omit('measure', props)}
+              tree={tree}
+              node={node}
+              theme={theme}
+              generationsCollapse={generationsCollapse}
+            />
+          </div>
+        </div>
+        {F.view(collapse) &&
+          contentRect.entry.height > innerHeightLimit &&
+          !!node.tags.length && (
+            <div style={{ minHeight: 10 }}>
+              <ExpandArrow collapse={collapse} tagsLength={node.tags.length} />
+            </div>
+          )}
       </div>
-    </div>
-    {F.view(collapse) &&
-      contentRect.entry.height > innerHeightLimit &&
-      !!props.node.tags.length && (
-        <div style={{ minHeight: 10 }}>
-          <ExpandArrow
-            collapse={collapse}
-            tagsLength={props.node.tags.length}
-          />
+      <Flex>
+        <div
+          style={
+            !F.view(generationsCollapse)
+              ? {
+                  width: '100%',
+                  marginTop: 20,
+                  position: 'relative',
+                  borderTop: '2px solid #EBEBEB',
+                }
+              : { display: 'none' }
+          }
+        >
+          {node.isStale && (
+            <Loader style={{ textAlign: 'center' }} loading={true}>
+              Loading...
+            </Loader>
+          )}
+          {!node.isStale &&
+            _.map((word) => (
+              <theme.Tag
+                tree={tree}
+                node={node}
+                onClick={({ value, label }) =>
+                  (e) => {
+                    tree.mutate(node.path, {
+                      ...(_.flow(
+                        _.find({ value: _.trim(value) }),
+                        _.isUndefined
+                      )(node.tags) && {
+                        tags: [
+                          ...node.tags,
+                          { word: value, distance: 3, label },
```

Comment:
We should factor out a method for this so that `distance: 3` isn't in multiple places